curriculum/challenges/english/blocks/workshop-shortest-path-algorithm/687b3926419dec576850b814.md

```diff
@@ -0,0 +1,81 @@
+---
+id: 687b3926419dec576850b814
+title: Step 19
+challengeType: 20
+dashedName: step-19
+---
+
+# --description--
+
+Now that you've calculated the new possible distance to the neighbor, check if it's better than the one currently stored in the `distances` list. If it is, update the distance.
+
+Inside the existing `if` block, add an `if` statement that checks if `new_distance` is less than `distances[node_no]`.
+
+Inside this new conditional block, update `distances[node_no]` to store the `new_distance`. 
+
+# --hints--
+
+You should have a nested `if` statement that checks if `new_distance` is less than `distances[node_no]`.
+
+```js
+({
+  test: () => {
+    assert(runPython(`_Node(_code).find_function("shortest_path").has_stmt("if new_distance < distances[node_no]:")
```

Comment:
the syntax error is from here I believe, as python expect a new line and indented content inside an if statement